### ISSUE TITLE
AnkiDroid Sync Error

### ISSUE THREAD

**Issue by danielelaclac** [ankidroid/Anki-Android#9540]
Everytime I try to sync the Ankidroid I get the same error as the image. I do not have a SD card. AnkiDroid sync is in the smartphone storage. Ive never had this error before. I use anki since 2020 february.

AnkiDroid Mobile Sync Error 

AnkiDroid Version = 2.15.6

Android Version = 10

Manufacturer = samsung

Model = SM-N9600

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 10; SM-N9600 Build/QP1A.190711.020; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/93.0.4577.82 Mobile Safari/537.36

ACRA UUID = 88a124e4-d2bb-438d-b823-1782aafd9db6

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true

![Screenshot_20210922-070126_AnkiDroid](https://user-images.githubusercontent.com/90069013/134352233-41354ec1-3f76-460e-b155-e3a9b7ca6fb4.jpg)

![Screenshot_20210922-065701_AnkiDroid](https://user-images.githubusercontent.com/90069013/134352283-75faeb17-e142-479c-bbb8-6b871b850368.jpg)

**Comment by mikehardy** [comment]
Hi there! Perhaps check app permissions and make sure "external storage" is granted to the app. The path looks fine, it just looks like AnkiDroid can't see it for some reason

**Comment by danielelaclac** [comment]
Already cheked. I have no sd card in the smartphone.

Em qua., 22 de set. de 2021 às 11:08, Mike Hardy ***@***.***>
escreveu:

**Comment by david-allison-1** [comment]
@danielelaclac 

Please sync to obtain the error message, then go to "Help - Get Help - Send Troubleshooting Report". 
We can help you further after this is submitted.

**Comment by mikehardy** [comment]
I was most interested in storage permission. I am very curious to know the app definitely has read/write external storage permission granted via system settings for the app.

This is unrelated to whether you actually have an SD card or not

**Comment by danielelaclac** [comment]
This has been already submitted.

Em qua, 22 de set de 2021 11:50, David Allison ***@***.***>
escreveu:

**Comment by danielelaclac** [comment]
I see no sd cards option. Is it what you are talking about?

Em qua, 22 de set de 2021 12:01, Mike Hardy ***@***.***>
escreveu:

**Comment by david-allison-1** [comment]
I see the reports, but there's nothing in the logs to help. 

Can you ensure that you submit the report immediately after seeing the error message that is in the main post.

Additional debug context:

https://github.com/ankidroid/Anki-Android/blob/27dbaeed881d3a1ef7b775311dabdf0780d3fc7f/AnkiDroid/src/main/java/com/ichi2/libanki/sync/FullSyncer.java#L94-L111

**Comment by danielelaclac** [comment]
Sent another report. Have u recived it? Thanks

Em qua, 22 de set de 2021 12:32, David Allison ***@***.***>
escreveu:

**Comment by david-allison-1** [comment]
I only see two from two hours ago for [88a124e4-d2bb-438d-b823-1782aafd9db6](https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/reports-browser/user/88a124e4-d2bb-438d-b823-1782aafd9db6)

**Comment by danielelaclac** [comment]
Strange :(
Ive sent another just 5min ago
AnkiDroid Version = 2.15.6

Android Version = 10

Manufacturer = samsung

Model = SM-N9600

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 10; SM-N9600
Build/QP1A.190711.020; wv) AppleWebKit/537.36 (KHTML, like Gecko)
Version/4.0 Chrome/93.0.4577.82 Mobile Safari/537.36

ACRA UUID = 76dc5cfe-f15e-4177-a965-0742701be98a

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true


Em qua, 22 de set de 2021 12:50, David Allison ***@***.***>
escreveu:

**Comment by danielelaclac** [comment]
Any news? :(  thanks

Em qua, 22 de set de 2021 12:01, Mike Hardy ***@***.***>
escreveu:

**Comment by mikehardy** [comment]
Not sure about David's analysis of crash reports but you have not answered my question about storage permission, so I have no news.

You said:

> I see no sd cards option. Is it what you are talking about?

I did not mean sd cards. I said storage. If you use your android system settings app to pull up the installed apps, and you select ankidroid and then go into it's permission settings, you should see an item for storage (or "Files and Media" on really new android versions) permission.

You must grant that permission for AnkiDroid to work, and I'm curious to hear if the permission is actually granted and shows up as granted in system settings

**Comment by danielelaclac** [comment]
I have sent you the pictures and asked you if that was the answer you were
seraching for

Em qua, 22 de set de 2021 14:07, Mike Hardy ***@***.***>
escreveu:

**Comment by mikehardy** [comment]
Please stop using mobile email replies. Click on the link to the actual github issue, and reply in here. Tell me what you see in the pictures. I do not see anything about system preferences in the pictures. You are showing me pictures from AnkiDroid settings, not Android settings.

**Comment by danielelaclac** [comment]
Here are the pics of the Android Settings
![photo_2021-09-22_14-17-21](https://user-images.githubusercontent.com/90069013/134390936-8a556971-8b47-4782-8ad6-2226571ba19a.jpg)
![photo_2021-09-22_14-17-26](https://user-images.githubusercontent.com/90069013/134390941-6f565692-e72b-4b6b-8181-2e71d1a81edb.jpg)
![photo_2021-09-22_14-17-33](https://user-images.githubusercontent.com/90069013/134390943-07eac140-3ae7-47af-bec5-44008abfed5d.jpg)
![photo_2021-09-22_14-17-40](https://user-images.githubusercontent.com/90069013/134390944-dd1bc533-5485-4be7-ac49-60e1fe674724.jpg)
:

**Comment by mikehardy** [comment]
Thanks for those - the Memoria setting in Permissoes is the one I was really curious about. Permitido indicates it should be working. The only other thing I can think of is if you set the path to `/sdcard/AnkiDroid` maybe that helps, but in general at this point I am stumped - it should be working.

**Comment by danielelaclac** [comment]
Same error:
![photo_2021-09-22_14-36-42](https://user-images.githubusercontent.com/90069013/134393765-2319fe57-8504-46c9-bfd0-491999428438.jpg)
![photo_2021-09-22_14-36-46](https://user-images.githubusercontent.com/90069013/134393767-12ab568e-148e-4774-a7bd-d1342c39251a.jpg)
![photo_2021-09-22_14-36-52](https://user-images.githubusercontent.com/90069013/134393771-3d0012cb-921e-485b-89d4-4f5cb2d536ab.jpg)
![photo_2021-09-22_14-36-57](https://user-images.githubusercontent.com/90069013/134393773-c7a96aa0-c281-47ef-8846-31089f87a6d0.jpg)

**Comment by danielelaclac** [comment]
I'm Anki user since 2020. Never had any problems. Now it has been 3 days I cant get acces to AnkiMobile. :(

**Comment by meepmerp** [comment]
I'm having the exact same problem with multiple decks between 2 different separate profiles/ankiweb accounts. 
This user is also having the same problem: https://forums.ankiweb.net/t/problem-syncing-error-when-downloading-from-ankiweb/13580

It seems a recent update from ankiweb has broken ankidroid's sync ability given all the users having problems. The reason I say its ankidroid is because the ankiweb website still works fine. It's only syncing between ankidroid and ankiweb/desktop that seems to have these errors. 

If you import the deck by USB onto ankidroid and upload to ankiweb, it will give this error when downloading to desktop: ![image](https://user-images.githubusercontent.com/16547308/134431640-4ea7bb22-b188-4f37-a9d9-ab75085ae23b.png)

 conversely

If you import the ankidroid deck to anki desktop, it will give the errors above with either SD card errors, or other errors. Something I also noticed with this error is that the sync download will jump from 0 KB to some high number, before eventually giving the SD card error. I don't have an SD card in the phone either and the storage location is the default ankidroid folder in internal storage. 

Some help would be greatly appreciated!

**Comment by david-allison-1** [comment]
Thanks, looking into it. We haven't pushed an update in a while, so it may be an AnkiWeb change.

**Comment by david-allison-1** [comment]
Okay, I get:

`Sync Error: Sync failed. Please visit ankiweb.net and log in as **email** to confirm the updated terms and conditions`

**Comment by david-allison-1** [comment]
@meepmerp @danielelaclac Could you try logging in to https://ankiweb.net and see if this resolves your issue.

**Comment by danielelaclac** [comment]
I have already logged in https://ankiweb.net. Now when I try to sync I get this other error. Any ideas?
![photo_2021-09-22_11-04-34](https://user-images.githubusercontent.com/90069013/134434600-d8018c26-cd05-4c41-911c-92dd27acf2a3.jpg)
:

**Comment by meepmerp** [comment]
I can't find a place to confirm the updated terms and conditions in my 2 accounts, but I did try to log out and in with both accounts on ankiweb. Same errors when trying to sync. 

I also tried to create a new ankiweb account and upload my desktop profile to ankiweb, which works fine (i'm able to answer on ankiweb and sync back to anki desktop). But the new account has the same behavior and error when trying to sync to ankidroid.

**Comment by danielelaclac** [comment]
"I also tried to create a new ankiweb account and upload my desktop profile to ankiweb, which works fine (i'm able to answer on ankiweb and sync back to anki desktop). But the new account has the same behavior and error when trying to sync to ankidroid."

The same has happened to me. I've created 3 new accounts. Have reinstalled AnkiDesktop in multiple pcs and AnkiMobile in mutiples devices. Nothing has worked. Have restored my cellphone and loose all data. AnkiMobile still not syncing. It made me very worried about using AnkiMobile for my tests.

**Comment by mikehardy** [comment]
Just to be clear on one point: AnkiMobile is the iOS version and is not related to us here in any way. We are AnkiDroid and work on Android. I think (based on your screenshots) this is just a name confusion and otherwise we are all talking about the same thing, but I want to be clear on that point in case you reach out for support in other channels - if you say AnkiMobile when you mean AnkiDroid it will be very confusing for people.

I'm looking into this as well - for me at least, so far syncing AnkiDroid -> AnkiWeb via incremental syncs is not a problem at least using 2.16alpha15 (current alpha release) but I have not tried AnkiDroid -> AnkiWeb -> Different AnkiDroid, nor have I tried throwing Anki Desktop in to the mix

**Comment by david-allison-1** [comment]
Anything on your side which would be relevant here?  @dae 

Can't reproduce these issues after accepting the new T&C

[Relevant code for "SD Card" error](https://github.com/ankidroid/Anki-Android/issues/9540#issuecomment-925043027) - likely downloading non-standard file content

**Comment by dae** [comment]
The 'must confirm terms to proceed' is a recent change, but that should result in a message to the user. The "error reading a body" issue looks like network troubles, and it may be worth checking for packet loss: https://faqs.ankiweb.net/syncing-failed-connection-timed-out-errors-when-syncing.html. The SD card issue is presumably something AnkiDroid-specific.

**Comment by danielelaclac** [comment]
I still get the same SD error. There is no file like this "collection" file
in my Ankidroid folder

Em qua, 22 de set de 2021 21:44, Damien Elmes ***@***.***>
escreveu:

**Comment by meepmerp** [comment]
I only get that error reading a body issue when I import my anki desktop .apkg export, fully sync it with ankiweb, and then try to sync ankidesktop with ankiweb. 

When I sync between anki desktop and ankiweb everything is fine; I can answer cards on either ankiweb or desktop and it'll save/sync properly. It's only when I try to sync to ankidroid that I get these SD card errors. I have never had an SD card in this phone for 4 years using ankidroid. This all started yesterday. I did also factory reset my phone and delete everything to try and get it working to no avail.

**Comment by danielelaclac** [comment]
"I have never had an SD card in this phone for 4 years using ankidroid.
This all started yesterday. I did also factory reset my phone and delete
everything to try and get it working to no avail."

Me too

Em qua, 22 de set de 2021 21:58, meepmerp ***@***.***>
escreveu:

**Comment by meepmerp** [comment]
Just throwing this out there, the new anki desktop version 2.1.48 did just get released 2 days ago. Might be a coincidence, but it it is around the same time this started, if they made some changes to how things are stored on ankiweb or such. 

Let me know if you want any more or my files or anything and thanks for looking into it.

**Comment by mikehardy** [comment]
I have not tried anything with apkg export/import, but with AnkiDroid and either AnkiDesktop 2.1.48 or 2.1.42 (tried first, then updated) I have no issues in any directions, so at least for now I can't reproduce. I won't have time to try .apkg export/import to specifically trigger everything, sorry

**Comment by meepmerp** [comment]
This morning around 10am the issue spontaneously fixed itself for both profiles/decks. I didn't do anything different, but I'm not complaining. Thanks for all the help.

**Comment by MmZy1** [comment]
@mikehardy 
Hi,
I have the same problem whenever my internet connection is around 60 kb/s max and it not associated with deck size and if it contains media or no, but if I full sync in 400 kb/s connection it's work fine, so I think it's a serious issue if users who have a bad connection internet need to do a full sync in ankidroid.
Thanks

**Comment by joaquintoral** [comment]
I also had this issue jsut now and spent my entire afternoon trying to fix it. I was finally able to fix the issue after deleting all anki files except collection on my android phone and do a sync. Despite doing this, the sync failed multiple times giving that exact SD card error despite me not having any SD card. It started when I deleted a card type and forced a one way sync from desktop to cloud. When I synced it to my phone, that SD card error just started appearing and it seemed as if there was no other way to fix it but to delete anki files and just do it again. I made the mistake of including the collection.media folder. Good thing I can simply copy and paste a compressed version of it from desktop to android phone to speed up the process as downloading the media files takes 10,000 years. Hopefully this won't happen again. I turned on my error reporting so hopefully the devs can pick it up.
